service/worker/deployment/deployment_name_workflow.go

```diff
@@ -0,0 +1,81 @@
+// The MIT License
+//
+// Copyright (c) 2024 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2024 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package deployment
+
+import (
+	sdkclient "go.temporal.io/sdk/client"
+	sdklog "go.temporal.io/sdk/log"
+	"go.temporal.io/sdk/workflow"
+)
+
+type (
+	DeploymentNameWorkflowArgs struct {
```

Comment:
It seems @dnr is using protobuf for this https://github.com/temporalio/temporal/blob/f4567a78978fc197966600fa89ba5effa880c52b/service/frontend/workflow_handler.go#L3008

Reply:
I spent quite sometime during the start of this project, if you may, thinking about whether to use protobuf or to use normal structs as parameters. I noticed for schedules, as you pointed out, we used protobuf but for batchers, we used structures. Both are internal workflows as well - I then tried looking for any recommended advice online as to what one should normally follow but did not find something concrete enough to chose one or the other. 

Keen on hearing what David thinks

Reply:
Thank you for jotting these down, David. The only thing that seems to bug me is the point you made about the size.

In both these workflows, we are aiming to use `CAN` where the size of the state (list holding DeploymentTaskQueues) shall grow. While we shall enforce limits as to how big the list can eventually get, I think this falls under the point you made about "use protos if messages are large".

I am too leaning towards proto and shall make the changes. 

*NOTE* - We can ignore this PR since the base branch was changed. A new one was opened with the right base branch and the changes will be present there. Apologies for the confusion :( 

new PR - https://github.com/temporalio/temporal/pull/6796